### ISSUE TITLE
Unexpected FlowExpression error

### ISSUE THREAD

**Issue by aagarwal1012** [typetools/checker-framework#2407]
I got the `Unexpected FlowExpression` error while running the following code:
[**Link**](http://eisop.uwaterloo.ca/live/#mode=display&code=import+org.checkerframework.checker.nullness.qual.Nullable%3B%0Aimport+org.checkerframework.checker.nullness.qual.NonNull%3B%0Aimport+org.checkerframework.checker.nullness.qual.RequiresNonNull%3B%0Aclass+A+%7B%0A++%0A++%40NonNull+String+message+%3D+%22default+string%22%3B%0A++%0A++%40RequiresNonNull(%22%231%22)%0A++void+setMessage(String+message)%7B%0A++++this.message+%3D+message%3B%0A++%7D%0A++%0A++String+getMessage()%7B%0A++++return+message%3B%0A++%7D%0A++%0A++void+main()%7B%0A++++A+object+%3D+new+A()%3B%0A++++object.setMessage(object.getMessage()+%2B+%22extra%22)%3B%0A++%7D+%0A%7D&typeSystem=nullness) to the code.
```java
import org.checkerframework.checker.nullness.qual.Nullable;
import org.checkerframework.checker.nullness.qual.NonNull;
import org.checkerframework.checker.nullness.qual.RequiresNonNull;
class A {
  
  @NonNull String message = "default string";
  
  @RequiresNonNull("#1")
  void setMessage(String message){
    this.message = message;
  }
  
  String getMessage(){
    return message;
  }
  
  void main(){
    A object = new A();
    object.setMessage(object.getMessage() + "extra");
  } 
}
```
The exact error message follows :
 I had used the Checker Framework Live Demo webpage to produce the error.
```bash
Error: Unexpected FlowExpression: ? (class org.checkerframework.dataflow.analysis.FlowExpressions$Unknown)
   Compilation unit: /A.java
   Last visited tree at line 19 column 45:
       object.setMessage(object.getMessage() + "extra");
   Exception: java.lang.Throwable; Stack trace: org.checkerframework.javacutil.BugInCF.<init>(BugInCF.java:14)
   org.checkerframework.framework.flow.CFAbstractStore.getValue(CFAbstractStore.java:463)
   org.checkerframework.common.basetype.BaseTypeVisitor.checkPreconditions(BaseTypeVisitor.java:1131)
   org.checkerframework.common.basetype.BaseTypeVisitor.visitMethodInvocation(BaseTypeVisitor.java:1016)
   org.checkerframework.common.basetype.BaseTypeVisitor.visitMethodInvocation(BaseTypeVisitor.java:159)
   com.sun.tools.javac.tree.JCTree$JCMethodInvocation.accept(JCTree.java:1477)
   com.sun.source.util.TreePathScanner.scan(TreePathScanner.java:68)
   org.checkerframework.framework.source.SourceVisitor.scan(SourceVisitor.java:91)
   org.checkerframework.common.basetype.BaseTypeVisitor.scan(BaseTypeVisitor.java:272)
   org.checkerframework.common.basetype.BaseTypeVisitor.scan(BaseTypeVisitor.java:159)
   com.sun.source.util.TreeScanner.visitExpressionStatement(TreeScanner.java:243)
   com.sun.tools.javac.tree.JCTree$JCExpressionStatement.accept(JCTree.java:1302)
   com.sun.source.util.TreePathScanner.scan(TreePathScanner.java:68)
   org.checkerframework.framework.source.SourceVisitor.scan(SourceVisitor.java:91)
   org.checkerframework.common.basetype.BaseTypeVisitor.scan(BaseTypeVisitor.java:272)
   org.checkerframework.common.basetype.BaseTypeVisitor.scan(BaseTypeVisitor.java:159)
   com.sun.source.util.TreeScanner.scanAndReduce(TreeScanner.java:81)
   com.sun.source.util.TreeScanner.scan(TreeScanner.java:91)
   com.sun.source.util.TreeScanner.visitBlock(TreeScanner.java:162)
   com.sun.tools.javac.tree.JCTree$JCBlock.accept(JCTree.java:918)
   com.sun.source.util.TreePathScanner.scan(TreePathScanner.java:68)
   org.checkerframework.framework.source.SourceVisitor.scan(SourceVisitor.java:91)
   org.checkerframework.common.basetype.BaseTypeVisitor.scan(BaseTypeVisitor.java:272)
   org.checkerframework.common.basetype.BaseTypeVisitor.scan(BaseTypeVisitor.java:159)
   com.sun.source.util.TreeScanner.scanAndReduce(TreeScanner.java:81)
   com.sun.source.util.TreeScanner.visitMethod(TreeScanner.java:144)
   org.checkerframework.framework.source.SourceVisitor.visitMethod(SourceVisitor.java:109)
   org.checkerframework.common.basetype.BaseTypeVisitor.visitMethod(BaseTypeVisitor.java:585)
   org.checkerframework.checker.initialization.InitializationVisitor.visitMethod(InitializationVisitor.java:322)
   org.checkerframework.checker.initialization.InitializationVisitor.visitMethod(InitializationVisitor.java:51)
   com.sun.tools.javac.tree.JCTree$JCMethodDecl.accept(JCTree.java:800)
   com.sun.source.util.TreePathScanner.scan(TreePathScanner.java:68)
   org.checkerframework.framework.source.SourceVisitor.scan(SourceVisitor.java:91)
   org.checkerframework.common.basetype.BaseTypeVisitor.scan(BaseTypeVisitor.java:272)
   org.checkerframework.common.basetype.BaseTypeVisitor.scan(BaseTypeVisitor.java:159)
   com.sun.source.util.TreeScanner.scanAndReduce(TreeScanner.java:81)
   com.sun.source.util.TreeScanner.scan(TreeScanner.java:91)
   com.sun.source.util.TreeScanner.scanAndReduce(TreeScanner.java:99)
   com.sun.source.util.TreeScanner.visitClass(TreeScanner.java:133)
   org.checkerframework.framework.source.SourceVisitor.visitClass(SourceVisitor.java:97)
   org.checkerframework.common.basetype.BaseTypeVisitor.processClassTree(BaseTypeVisitor.java:350)
   org.checkerframework.checker.initialization.InitializationVisitor.processClassTree(InitializationVisitor.java:280)
   org.checkerframework.common.basetype.BaseTypeVisitor.visitClass(BaseTypeVisitor.java:310)
   org.checkerframework.common.basetype.BaseTypeVisitor.visitClass(BaseTypeVisitor.java:159)
   com.sun.tools.javac.tree.JCTree$JCClassDecl.accept(JCTree.java:720)
   com.sun.source.util.TreePathScanner.scan(TreePathScanner.java:50)
   org.checkerframework.framework.source.SourceVisitor.visit(SourceVisitor.java:85)
   org.checkerframework.framework.source.SourceChecker.typeProcess(SourceChecker.java:974)
   org.checkerframework.common.basetype.BaseTypeChecker.typeProcess(BaseTypeChecker.java:490)
   org.checkerframework.javacutil.AbstractTypeProcessor$AttributionTaskListener.finished(AbstractTypeProcessor.java:182)
   com.sun.tools.javac.api.ClientCodeWrapper$WrappedTaskListener.finished(ClientCodeWrapper.java:681)
   com.sun.tools.javac.api.MultiTaskListener.finished(MultiTaskListener.java:111)
   com.sun.tools.javac.main.JavaCompiler.flow(JavaCompiler.java:1342)
   com.sun.tools.javac.main.JavaCompiler.flow(JavaCompiler.java:1296)
   com.sun.tools.javac.main.JavaCompiler.compile2(JavaCompiler.java:901)
   com.sun.tools.javac.main.JavaCompiler.compile(JavaCompiler.java:860)
   com.sun.tools.javac.main.Main.compile(Main.java:523)
   com.sun.tools.javac.api.JavacTaskImpl.doCall(JavacTaskImpl.java:129)
   com.sun.tools.javac.api.JavacTaskImpl.call(JavacTaskImpl.java:138)
   traceprinter.ramtools.CompileToBytes.compileFiles(CompileToBytes.java:55)
   traceprinter.ramtools.CompileToBytes.compileFile(CompileToBytes.java:37)
   checkerprinter.InMemory.<init>(InMemory.java:143)
   checkerprinter.InMemory.main(InMemory.java:82)
```
This error message won't appear when _@RequiresNonNull_ is removed and instead `@NonNull` is used.

**Comment by wmdietl** [comment]
Thanks for the test case!
Somewhat simplified:

```java
import org.checkerframework.checker.nullness.qual.RequiresNonNull;

class A {
  @RequiresNonNull("#1")
  void setMessage(String message) {}

  void main() {
    new A().setMessage("foo" + "bar");
  } 
}
```

We should discuss whether pre-condition annotations should ever be allowed to use only parameter flow expressions. It would be more ideomatic to simply add the corresponding type annotation on the parameter.
However, such a crash shouldn't happen.
